src/slave/slave.cpp

```diff
@@ -3654,6 +3655,32 @@ void Slave::launchExecutor(
       authenticationToken.get(),
       framework->info.checkpoint());
 
+
+  if(environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled"){
+
+    // Slave has CNI enabled but doesn't define any networks, public and private subnets must be injected.
+    ContainerInfo *containerInfo = containerConfig.mutable_container_info();
+
+    NetworkInfo *privateNet = containerInfo->add_network_infos();
```

Comment:
I think we should consider now the opportunity to have only one subnet (as discussed on slack).

Reply:
I could take the time to run some side experiments once this is running real demo apps.

Reply:
I second this, having one subnet would simplify this PR, but likely also future changes in the frameworks

Reply:
should this stay pending until we simplify the stack then?

Reply:
Why not simplifying the stack now?